### PR TITLE
lb/Consolidate gias school and school models

### PR DESCRIPTION
**Pull request by elceebee**
## Context

Originally, we had a school record joined to the GiasSchool record. In practice, this is becoming unweildy as every detail we need for a school needs to be delegated to the GiasSchool

## Changes proposed in this pull request

- Adds all the `GiasSchool` columns, validations, and pg_search scopes to the `School` model
- The `GiasCsvImporter` upserts School records
- `School.all` returns all GIAS schools
- The `GiasSchool` model and db table have been removed
- Seeds have been updated
- Adds Faker gem so not all our schools are named Hogwarts
- Updates erd document to reflect changes 

## Guidance to review

If we were in production, I would have done this in a couple PRs, but since we're not...
Two things to test: 
- run migrations, then run the `rake gias_update` and make sure everything works
- like a new developer on the team, drop / create / migrate and seed the database and make sure everything works

## Link to Trello card

https://trello.com/c/3Ke94xwX

## Things to check

- [NA] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request -- It does, but since we're not in production, that's ok!
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [NA] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [NA] API release notes have been updated if necessary
- [NA] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [NA] Required environment variables have been updated or [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

## Screenshots

No screen shots etc -- everything should just work as it did before the changes.

### CODE REVIEW

**Comment by elceebee** [comment]
[Consolidate GiasSchool and School models](https://trello.com/c/3Ke94xwX/99-consolidate-giasschool-and-school-models)

**Review by CatalinVoineag — APPROVED** [review]
Pulled this on local and ran the migrations on my existing db but also after dropping it. Works as expected, very nice work!

File: Gemfile [hunk 1]
```diff
@@ -107,6 +107,7 @@ group :test, :development do
   gem "dotenv-rails"
   gem "erb_lint", require: false
   gem "factory_bot_rails"
+  gem "faker"
```
Comment (ollietreend) [inline Gemfile:110]: Just a heads up: PR #92 also adds Faker. So depending on who merges first, there _may_ be a merge conflict here.

File: spec/factories/schools.rb [hunk 2]
```diff
@@ -18,6 +47,7 @@
 FactoryBot.define do
   factory :school do
     sequence(:urn) { _1 }
+    name { Faker::Educator.primary_school }
```
Comment (ollietreend) [inline spec/factories/schools.rb:50]: 🤩

File: docs/data-model.md [hunk 3]
```diff
@@ -110,26 +102,26 @@ Additionally, Providers will need to be onboarded to use the School Placements s
 
 ### Schools
 
-An 'onboarded' School will have a record in the `schools` table (the `School` entity in our ERD).
+All schools from the gias import will have a record in the `schools` table (the `School` entity in our ERD).
 
-The `placements_enabled` and `claims_enabled` boolean attributes will indicate which service(s) the School has been onboarded into. It's possible for Schools to be onboarded to only one service, or both.
+The `placements` and `claims` boolean attributes will indicate which service(s) the School has been onboarded into. It's possible for Schools to be onboarded to only one service, or both.
```
Comment (ollietreend) [inline docs/data-model.md:107]:
```suggestion
The `placements` and `claims` boolean attributes will indicate which service(s) the School has been onboarded into. It's possible for Schools to be onboarded to both services, one service, or neither service.
```

File: docs/data-model.md [hunk 4]
```diff
@@ -110,26 +102,26 @@ Additionally, Providers will need to be onboarded to use the School Placements s
...
+| 100001 | School B | 0          | 1      |
+| 100002 | School C | 1          | 1      |
+| 100003 | School D | 0          | 0      |
 
 - School A has only been onboarded into the School Placements service.
 - School B has only been onboarded into the Track & Pay service.
 - School C has been onboarded into both services.
+- School D is a school from the Gias import, but it is not onboarded into either service
 
 If a School doesn't exist in the `schools` table, it hasn't been onboarded to either service yet.
```
Comment (ollietreend) [inline docs/data-model.md:124]:
```suggestion
```

File: docs/data-model.md [hunk 5]
```diff
@@ -110,26 +102,26 @@ Additionally, Providers will need to be onboarded to use the School Placements s
 
 ### Schools
 
-An 'onboarded' School will have a record in the `schools` table (the `School` entity in our ERD).
+All schools from the gias import will have a record in the `schools` table (the `School` entity in our ERD).
```
Comment (ollietreend) [inline docs/data-model.md:105]: Let's try to stylise it as GIAS wherever possible. That's what's used on [the service's about page](https://get-information-schools.service.gov.uk/about).

```suggestion
All schools from the GIAS import will have a record in the `schools` table (the `School` entity in our ERD).
```

File: docs/data-model.md [hunk 6]
```diff
@@ -110,26 +102,26 @@ Additionally, Providers will need to be onboarded to use the School Placements s
...
+|--------|----------|------------|--------|
+| 100000 | School A | 1          | 0      |
+| 100001 | School B | 0          | 1      |
+| 100002 | School C | 1          | 1      |
+| 100003 | School D | 0          | 0      |
 
 - School A has only been onboarded into the School Placements service.
 - School B has only been onboarded into the Track & Pay service.
 - School C has been onboarded into both services.
+- School D is a school from the Gias import, but it is not onboarded into either service
```
Comment (ollietreend) [inline docs/data-model.md:121]:
```suggestion
- School D is a school from the GIAS import, but it is not onboarded into either service
```

File: spec/system/claims/view_organisations_spec.rb [hunk 7]
```diff
@@ -37,7 +37,7 @@ def and_persona_has_multiple_schools(persona)
   end
 
   def and_persona_has_one_school(persona)
-    create(:membership, user: persona, organisation: create(:school))
+    create(:membership, user: persona, organisation: create(:school, name: "Hogwarts"))
```
Comment (ollietreend) [inline spec/system/claims/view_organisations_spec.rb:40]: I'm glad we didn't lose our Harry Potter reference. It lives on! 😆

**Review by ollietreend — CHANGES_REQUESTED** [review]
Really nice work! ✨🚀 This is looking super.

Mega brownie points for remembering to update `docs/data-model.md` too! ⭐

The only changes I've requested are wording tweaks in the data model doc.

File: Gemfile (hunk 1 above)
Comment (elceebee) [reply Gemfile:110]: It's a race!

**Review by ollietreend — APPROVED** [review]

**Pull request merged**